### PR TITLE
[ui-core] add useSaveData hook for api calls

### PR DESCRIPTION
**Pull request by ramprasadagarwal**
## What changes were proposed in this pull request?

- Add useSaveData hook for react to save the data to the backend. This is similar to useLoadData hook implemented for fetching data.

## How was this patch tested?

- Added unit tests

## Usage

```
const { loading: isSaving, save } = useSaveData(API_URL, {
  onSuccess: () => {
    setIsEditing(false);
  }
});

const handleSave = () =>
  save({
    path: fileData.path
  });
```

### CODE REVIEW

**Review by nidhibhatg — APPROVED** [review]
LGTM!

File: desktop/core/src/desktop/js/utils/hooks/useSaveData.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      // Avoid Posting data if the skip option is true
+      // or if the URL is not provided
+      if (options?.skip || !url) {
+        return;
+      }
+      setLoading(true);
+      setError(undefined);
+
+      try {
+        const PostUrl = localOptions?.urlPrefix ? `${localOptions.urlPrefix}${url}` : url;
```
Comment (bjornalm) [inline desktop/core/src/desktop/js/utils/hooks/useSaveData.ts:62]: Wouldn't part of the reason for creating a save hook be to hide away details like url-prefix etc? Is there a use case for using this with different url-prefixes in Hue?

**Review by bjornalm — APPROVED** [review]
LGTM, but see question.

File: desktop/core/src/desktop/js/utils/hooks/useSaveData.ts (hunk 1 above)
Comment (ramprasadagarwal) [reply desktop/core/src/desktop/js/utils/hooks/useSaveData.ts:62]: Agree, I believe I have over complicated it. Let me remove the urlPrefix here and also in useLoadData

**Pull request merged**